### PR TITLE
feat: added layer into an existing best implementaion 

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                     |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: config/default.json [hunk 1]
```diff
@@ -12,12 +12,12 @@
     "port": "8080"
   },
   "mapproxy": {
-    "yamlFilePath": "tests/unit/mock/mockContent.yaml",
+    "yamlFilePath": "/home/shlomikoncha/Desktop/mapproxy/mapproxy.yaml",
```
Comment (asafMasa) [inline config/default.json:15]: revert the file path

File: openapi3.yaml [hunk 2]
```diff
@@ -52,7 +53,7 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/error'
-  /best:
+  /layer/best:
```
Comment (asafMasa) [inline openapi3.yaml:56]: we said we will remove the term "best" from our code, no?

File: src/common/validations/isLayerNameExists.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { IMapProxyJsonDocument } from '../interfaces';
+
+export function isLayerNameExists(jsonDocument: IMapProxyJsonDocument, layerName: string): boolean {
```
Comment (asafMasa) [inline src/common/validations/isLayerNameExists.ts:3]: if the method name is "isLayerNameExists" it should return false it the layer doesn't exit not throw an exception

For that you should choose other method name such as validate....

File: src/layers/controllers/layersController.ts [hunk 4]
```diff
@@ -2,11 +2,12 @@ import { RequestHandler } from 'express';
 import httpStatus from 'http-status-codes';
 import { injectable, inject } from 'tsyringe';
 import { Services } from '../../common/constants';
-import { ILogger, ILayerPostRequest } from '../../common/interfaces';
+import { ILogger, ILayerPostRequest, ILayerToBestRequest } from '../../common/interfaces';
 import { LayersManager } from '../models/layersManager';
 
 type CreateLayerHandler = RequestHandler<undefined, ILayerPostRequest, ILayerPostRequest>;
 type GetLayerHandler = RequestHandler<undefined, ILayerPostRequest>;
+type CreateBestHandler = RequestHandler<undefined, ILayerToBestRequest, ILayerToBestRequest>;
```
Comment (asafMasa) [inline src/layers/controllers/layersController.ts:10]: we said we will remove the term "best" from our code, no?

File: src/layers/controllers/layersController.ts [hunk 5]
```diff
@@ -24,4 +25,13 @@ export class LayersController {
       next(error);
     }
   };
+
+  public addLayerToBest: CreateBestHandler = (req, res, next) => {
```
Comment (asafMasa) [inline src/layers/controllers/layersController.ts:29]: we said we will remove the term "best" from our code, no?

File: src/layers/models/layersManager.ts [hunk 6]
```diff
@@ -49,13 +60,24 @@ export class LayersManager {
...
   }
 
-  // check if requested layer name is already exists in mapproxy config file (layer name must be unique)
-  public isLayerNameExists(jsonDocument: IMapProxyJsonDocument, layerName: string): boolean {
-    try {
-      const publishedLayers: string[] = Object.keys(jsonDocument.caches);
-      return publishedLayers.includes(layerName);
-    } catch (error) {
-      throw new Error(error);
+  public addLayerToBest(layerToBestRequest: ILayerToBestRequest): void {
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:63]: we said we will remove the term "best" from our code, no?

File: src/layers/routes/layersRouterFactory.ts [hunk 7]
```diff
@@ -8,6 +8,7 @@ const layersRouterFactory: FactoryFunction<Router> = (dependencyContainer) => {
 
   router.get('/', controller.getLayer);
   router.post('/', controller.addLayer);
+  router.post('/best', controller.addLayerToBest);
```
Comment (asafMasa) [inline src/layers/routes/layersRouterFactory.ts:11]: we said we will remove the term "best" from our code, no?

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/common/validations/isLayerNameExists.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/validations/isLayerNameExists.ts:3]: it is returns flase it layer doesn't exists : 
 return publishedLayers.includes(layerName); => includes() returns true/false, the the error throw is from the try catch section

File: src/layers/routes/layersRouterFactory.ts (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/routes/layersRouterFactory.ts]: fixed in ee20559

File: src/layers/models/layersManager.ts (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts]: fixed in ee20559

File: src/layers/controllers/layersController.ts (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/controllers/layersController.ts]: fixed in ee20559

File: src/layers/controllers/layersController.ts (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/controllers/layersController.ts]: fixed in ee20559

File: openapi3.yaml (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml]: fixed in ee20559

File: config/default.json (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply config/default.json]: fixed in ee20559

File: src/bests/models/bestsManager.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+@injectable()
+export class BestsManager {
+  public constructor(
+    @inject(Services.LOGGER) private readonly logger: ILogger,
+    @inject(Services.MAPPROXY) private readonly mapproxyConfig: IMapProxyConfig
+  ) {}
+
+  public getBest(): IBestLayer {
+    this.logger.log('info', 'Get layer request');
+    const bestLayers: IBestLayer = { layers: ['mock', 'mock2', 'mock3'] };
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/bests/models/bestsManager.ts]: fixed in cefa67a

File: src/layers/models/layersManager.ts [hunk 9]
```diff
@@ -49,13 +60,24 @@ export class LayersManager {
...
 
-  // check if requested layer name is already exists in mapproxy config file (layer name must be unique)
-  public isLayerNameExists(jsonDocument: IMapProxyJsonDocument, layerName: string): boolean {
-    try {
-      const publishedLayers: string[] = Object.keys(jsonDocument.caches);
-      return publishedLayers.includes(layerName);
-    } catch (error) {
-      throw new Error(error);
+  public addLayerToMosaic(layerToMosaicRequest: ILayerToMosaicRequest): void {
+    this.logger.log('info', `Add layer: ${layerToMosaicRequest.layerName} to mosaic: ${layerToMosaicRequest.mosaicName} request`);
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:64]: is there a try catch for all o this method?

**Review by asafMasa — APPROVED** [review]

**Pull request merged**